Earthfile

```diff
@@ -12,48 +12,265 @@
+  # Install required software development tools and ROS tools (and vim included for convenience)
+  RUN --mount=type=cache,mode=0777,target=/var/cache/apt,sharing=locked,id=cache_apt_cache \
+      --mount=type=cache,mode=0777,target=/var/lib/apt,sharing=locked,id=lib_apt_cache \
+      apt-get update && \
+      apt-get install -y \
+        bison \
+        build-essential \
+        cmake \
+        git \
+        python3-colcon-common-extensions \
+        python3-flake8 \
+        python3-flake8-blind-except \
+        python3-flake8-builtins \
+        python3-flake8-class-newline \
+        python3-flake8-comprehensions \
+        python3-flake8-deprecated \
+        python3-flake8-docstrings \
+        python3-flake8-import-order \
+        python3-flake8-quotes \
+        python3-pip \
+        python3-pytest \
+        python3-pytest-cov \
+        python3-pytest-repeat \
+        python3-pytest-rerunfailures \
+        python3-rosdep \
+        python3-rosinstall-generator \
+        python3-setuptools \
+        python3-vcstool \
+        wget \
+        vim
+
+  # Define the username and key variables
+  ENV USERNAME spaceros-user
+  ENV HOME_DIR=/home/${USERNAME}
+  ENV SPACEROS_DIR=/opt/spaceros
+
+  # Create the spaceros directory
+  RUN mkdir --mode=777 -p ${SPACEROS_DIR}
+
+  # Create a spaceros user
+  RUN useradd -m ${USERNAME} && \
+    echo "${USERNAME}:${USERNAME}" | chpasswd && \
+    usermod --shell /bin/bash ${USERNAME} && \
+    usermod -aG sudo ${USERNAME} && \
+    mkdir -p /etc/sudoers.d && \
+    echo "${USERNAME} ALL=(ALL) NOPASSWD:ALL" >> /etc/sudoers.d/${USERNAME} && \
+    chmod 0440 /etc/sudoers.d/${USERNAME}
+  USER ${USERNAME}
+  WORKDIR ${SPACEROS_DIR}
+
+  # Update the OpenGL version
+  RUN sudo add-apt-repository ppa:kisak/kisak-mesa
+  RUN --mount=type=cache,mode=0777,target=/var/cache/apt,sharing=locked,id=cache_apt_cache \
+      --mount=type=cache,mode=0777,target=/var/lib/apt,sharing=locked,id=lib_apt_cache \
+      sudo apt update && sudo apt upgrade -y
+
+  # Create install location and copy in relevant scripts
+  COPY --chown ${USERNAME}:${USERNAME} --dir docker/scripts/ ${SPACEROS_DIR}
 
 repos-file:
   FROM +setup
-
-  # Copy inputs and generate a repos file
-  WORKDIR /root
-  COPY --dir scripts/ ./
   COPY excluded-pkgs.txt ./
   COPY spaceros-pkgs.txt ./
   COPY spaceros.repos ./
+  # This is a fresh image, so we do not need to exclude installed packages.
   RUN --no-cache sh scripts/generate-repos.sh \
                  --outfile ros2.repos \
                  --packages spaceros-pkgs.txt \
                  --excluded-packages excluded-pkgs.txt \
-                 --rosdistro $ROSDISTRO
+                 --exclude-installed false \
+                 --rosdistro ${ROS_DISTRO}
   RUN --no-cache python3 scripts/merge-repos.py ros2.repos spaceros.repos -o output.repos
-
-  # Save the generated .repos file
   SAVE ARTIFACT output.repos AS LOCAL ros2.repos
+
+spaceros-artifacts:
```

Comment:
Just so I know - is the problem that I raised in here https://github.com/eholum/space-ros/pull/1#discussion_r1681445294 fixed?